### ISSUE TITLE
Subscriptions are not triggered for primitive selectors on subsequent dispatches with same payload

### ISSUE THREAD

**Issue by ashkandaie** [ngrx/platform#208]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request
</code></pre>

## What is the current behavior?

When subscribing to a primitive selector such as a boolean, the first dispatched message triggers the subscription but subsequent dispatched messages with the same payload do not trigger the subscription. If you change the selector to the parent selector which contains the primitive data everything works as expected.

## Expected behavior:

All dispatched messages should trigger subscriptions for selectors that return complex objects or primitive selectors. 

## Minimal reproduction of the problem with instructions:

layout state: 
```javascript

export interface LayoutState {
  showSidenav: boolean;
}

```

layout actions:

```javascript
import { Action } from '@ngrx/store';

export const OPEN_SIDENAV = '[Layout] Open Sidenav';
export const CLOSE_SIDENAV = '[Layout] Close Sidenav';

export class OpenSidenavAction implements Action {
  readonly type = OPEN_SIDENAV;
}

export function openSidenav() {
  return new OpenSidenavAction();
}

export class CloseSidenavAction implements Action {
  readonly type = CLOSE_SIDENAV;
}

export function closeSidenav() {
  return new CloseSidenavAction();
}

export type Actions = OpenSidenavAction | CloseSidenavAction;

```

layout reducer:
```javascript
import { createSelector, createFeatureSelector } from '@ngrx/store';
import { LayoutState } from './state';
import * as layout from './actions';

const initialState: LayoutState = {
  showSidenav: false,
};

export function layoutReducer(state = initialState, action: layout.Actions): LayoutState {
  switch (action.type) {
    case layout.CLOSE_SIDENAV:
      return {
        showSidenav: false,
      };

    case layout.OPEN_SIDENAV:
      return {
        showSidenav: true,
      };

    default:
      return state;
  }
}

export const selectLayout = createFeatureSelector<LayoutState>('layout');

export const selectShowSidenav = (state: LayoutState) => state.showSidenav;

export const getShowSidenav = createSelector(selectLayout, selectShowSidenav);
```

subscription:

```javascript
showSidenav$: Observable<boolean>;

this.showSidenav$ = <any>this._store.select(fromLayout.getShowSidenav);   
  this.showSidenav$.subscribe((value) => {
  console.log('Value Changed', value);
});
```

dispatch (run on button click so you can dispatch same action / payload multiple times):
```javascript

this._store.dispatch(new layout.OpenSidenavAction());

```

You will notice that only the first dispatch is caught by the subscription. 

If you change the selector to the parent object like this:

```javascript

    this.showSidenav$ = <any>this._store.select(fromLayout.selectLayout);

```

All dispatches are processed by the subscription.
 
## Version of affected browser(s),operating system(s), npm, node and ngrx:

Chrome, MacOS, npm 5.3.0, Node 8.2.0 ngrx/store: 4.0.0, ngrx/router-store: 4.0.0 ngrx/effects: 4.0.1  

## Other information:

N/A

:heartpulse: ngrx? Please consider supporting our collective: 👉  [donate](https://opencollective.com/ngrx/donate)

**Comment by bharathmuppa** [comment]
I wonder why it is closed as it is not fixed yet ? (even in #211)

**Comment by manklu** [comment]
@bharathmuppa

Instead of commenting old issues you should open a new one with a description and a reproduction.
